lib/model.js

```diff
@@ -1061,36 +1083,24 @@ Connection.from = (url, callback) => {
 
   const unescapedUrl = unescape(url);
 
-  parseConnectionString(unescapedUrl, (error, parsed) => {
-    if (error) {
-      return callback(error);
+  parseConnectionString(unescapedUrl, (parseError, parsed) => {
+    if (parseError) {
+      return callback(parseError);
     }
 
     let attrs = Object.assign(
       {},
       {
         hosts: parsed.hosts,
-        hostname: parsed.hosts[0].host,
-        port: parsed.hosts[0].port,
+        hostname: isSrvRecord ? parsed.srvHost : parsed.hosts[0].host,
```

Comment:
Here I use `parsed.srvHost` instead of the previous extra parsing `URL.parse(unescapedUrl, true);`, that was used to get the original hostname because the driver did not return this info before and adds cluster information to the hostname after resolving DNS hosts.